flaky test TestLoadDataAutoRandomXXX

```
    dbtestkit.go:71: 
        	Error Trace:	testkit/dbtestkit.go:71
        	            				server/server_test.go:524
        	            				server/server_test.go:151
        	            				server/server_test.go:516
        	            				server/tidb_serial_test.go:82
        	Error:      	Received unexpected error:
        	            	Error 1105 (HY000): runtime error: index out of range [0] with length 0
        	Test:       	TestLoadDataAutoRandom
        	Messages:   	sql:load data local infile "/tmp/load_data_txn_error.csv228170663" into table t (c2, c3), args:[]
```

the reason is

```
[2023/03/14 21:46:22.069 +08:00] [ERROR] [load_data.go:893] ["failed to get row"] [conn=2199023255957] [error="[kv:8024]invalid transaction"] [errorVerbose="[kv:8024]invalid transaction\ngithub.com/pingcap/errors.AddStack\n\texternal/com_github_pingcap_errors/errors.go:174\ngithub.com/pingcap/tidb/session.(*txnManager).ActivateTxn\n\tsession/txnmanager.go:260\ngithub.com/pingcap/tidb/session.(*session).Txn\n\tsession/session.go:2501\ngithub.com/pingcap/tidb/executor.(*InsertValues).allocAutoRandomID\n\texecutor/insert_common.go:1049\ngithub.com/pingcap/tidb/executor.(*InsertValues).adjustAutoRandomDatum\n\texecutor/insert_common.go:1012\ngithub.com/pingcap/tidb/executor.(*InsertValues).fillColValue\n\texecutor/insert_common.go:623\ngithub.com/pingcap/tidb/executor.(*InsertValues).fillRow\n\texecutor/insert_common.go:673\ngithub.com/pingcap/tidb/executor.(*InsertValues).getRow\n\texecutor/insert_common.go:567\ngithub.com/pingcap/tidb/executor.(*LoadDataWorker).parserData2TableData\n\texecutor/load_data.go:887\ngithub.com/pingcap/tidb/executor.(*LoadDataWorker).ReadOneBatchRows\n\texecutor/load_data.go:802\ngithub.com/pingcap/tidb/executor.(*LoadDataWorker).processStream\n\texecutor/load_data.go:580\ngithub.com/pingcap/tidb/executor.(*LoadDataWorker).Load.func2\n\texecutor/load_data.go:435\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\texternal/org_golang_x_sync/errgroup/errgroup.go:75\nruntime.goexit\n\tsrc/runtime/asm_amd64.s:1598"] [stack="github.com/pingcap/tidb/executor.(*LoadDataWorker).parserData2TableData\n\texecutor/load_data.go:893\ngithub.com/pingcap/tidb/executor.(*LoadDataWorker).ReadOneBatchRows\n\texecutor/load_data.go:802\ngithub.com/pingcap/tidb/executor.(*LoadDataWorker).processStream\n\texecutor/load_data.go:580\ngithub.com/pingcap/tidb/executor.(*LoadDataWorker).Load.func2\n\texecutor/load_data.go:435\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\texternal/org_golang_x_sync/errgroup/errgroup.go:75"]
```

--- a/executor/load_data.go
+++ b/executor/load_data.go
@@ -415,6 +415,11 @@ func (e *LoadDataWorker) Load(
 	// done is used to let commitWork goroutine notify UpdateJobProgress
 	// goroutine that the job is finished.
 	done := make(chan struct{})
+	// both processStream and commitWork goroutines will use this txn.
+	err = sessiontxn.NewTxn(ctx, e.Ctx)
+	if err != nil {
+		return err
+	}
 
 	// processStream goroutine.
 	group.Go(func() error {
@@ -441,6 +446,7 @@ func (e *LoadDataWorker) Load(
 	})
 	// commitWork goroutine.
 	group.Go(func() error {
+		failpoint.Inject("BeforeCommitWork", nil)
 		err2 := e.commitWork(groupCtx)
 		if err2 == nil {
 			close(done)
@@ -629,11 +635,6 @@ func (e *LoadDataWorker) commitWork(ctx context.Context) (err error) {
 			err = errors.Errorf("%v", r)
 		}
 	}()
-
-	err = sessiontxn.NewTxn(ctx, e.Ctx)
-	if err != nil {
-		return err
-	}
 
 	var (
 		tasks               uint64

--- a/server/tidb_serial_test.go
+++ b/server/tidb_serial_test.go
@@ -26,6 +26,7 @@ import (
 
 	"github.com/go-sql-driver/mysql"
 	"github.com/pingcap/errors"
+	"github.com/pingcap/failpoint"
 	"github.com/pingcap/tidb/config"
 	tmysql "github.com/pingcap/tidb/parser/mysql"
 	"github.com/pingcap/tidb/sessionctx/variable"
@@ -77,6 +78,12 @@ func TestConfigDefaultValue(t *testing.T) {
 // Fix issue#22540. Change tidb_dml_batch_size,
 // then check if load data into table with auto random column works properly.
 func TestLoadDataAutoRandom(t *testing.T) {
+	err := failpoint.Enable("github.com/pingcap/tidb/executor/BeforeCommitWork", "sleep(1000)")
+	require.NoError(t, err)
+	defer func() {
+		//nolint:errcheck
+		_ = failpoint.Disable("github.com/pingcap/tidb/executor/BeforeCommitWork")
+	}()
 	ts := createTidbTestSuite(t)
 
 	ts.runTestLoadDataAutoRandom(t)

